src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorDynamicFileInfoProvider.cs

```diff
@@ -20,22 +21,22 @@ internal class RazorDynamicFileInfoProvider : IDynamicFileInfoProvider
 
     private class ProvideDynamicFileParams
     {
-        [JsonPropertyName("razorFiles")]
-        public required Uri[] RazorFiles { get; set; }
+        [JsonPropertyName("document")]
+        public required TextDocumentIdentifier Document { get; set; }
     }
 
     private class ProvideDynamicFileResponse
     {
-        [JsonPropertyName("generatedFiles")]
-        public required Uri[] GeneratedFiles { get; set; }
+        [JsonPropertyName("document")]
```

Comment:
Nit: perhaps we should rename this just so the response and params are different. `csharpDocument` and/or `razorDocument` for the above?

Reply:
sgtm